### ISSUE TITLE
Add user/group changelogs

### ISSUE THREAD

**Issue by mcao2** [netbox-community/netbox#17882]
### NetBox version

v3.7.8

### Feature type

Change to existing functionality

### Triage priority

N/A

### Proposed functionality

Before NetBox v3.6.0, user/group management is done using django's admin page and we have log entries for related object changes. But starting from v3.6.0 (https://github.com/netbox-community/netbox/pull/12877), NetBox now manage user/group in its Admin tab, and django admin seems completely removed in v4.0. I propose we add similar change logging feature for user/group changes so that we can better understand/audit the activities. 

### Use case

When there's multiple admins or users with user/group change permissions, it becomes unmanageable to ask who made that change and when that change happened. If we can have a similar Admin LogEntry in NetBox then it would greatly improve the visibility for admins.

**Comment by PieterL75** [comment]
I think an audit log might be a better place.. That log could also include login/logout/loginattempts...

**Comment by mcao2** [comment]
> I think an audit log might be a better place.. That log could also include login/logout/loginattempts...

I completely agree, adjusted the issue title to not suggest where the changelog should exists

**Comment by bctiemann** [comment]
Another possibility might be to add the `ChangeLoggingMixin` to the `User` and `Group` models, which are already subclassed from `AbstractUser` and can support additional customizations.

A separate audit log is a good idea, but we already have pervasive change logging for many other models that we might as well extend to users.

**Comment by ITJamie** [comment]
This came up during the isp working group. The idea of a separate audit log was mentioned to track admin related changes with a bit more detail (ips used when making change) and also to be able to limit who can see the audit logs
